### PR TITLE
Feat: (block-editor-utils) Add RangeField

### PR DESCRIPTION
**Pull request by theodesp**
## Tasks

- [ ] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.
- [ ] If a code change, I have written testing instructions that the whole team & outside contributors can understand.
- [ ] I have written and included a comprehensive [changeset](https://github.com/wpengine/faustjs/blob/canary/DEVELOPMENT.md#deployment) to properly document the changes I've made.

## Description

Adds `RangeField` Control in block-editor-utils package.

## Related Issue(s):

https://github.com/wpengine/faustjs/issues/1522

## Testing
After pulling this branch down, run the following commands in the faustjs monorepo:
1. run `npm install` 
2. run `npm run build` 
3. run `npm run build -w packages/block-editor-utils`
4. run `cd packages/block-editor-utils`
5. run `npm link`

Switch to your testing site's repo.

6. On your WordPress instance create a new block plugin using [@wordpress/create-block](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-create-block/)
npx @wordpress/create-block hello-world-block
7. cd to your new block plugin's folder 
8. Inside the `hello-world-block` plugin folder, run `npm link @faustwp/block-editor-utils` to add a reference to the package.
9. Replace `hello-world-block`'s generated index.js file with the below code block:
```js
// In app/public/wp-content/plugins/hello-world-block/src/index.js:

import { addFilter } from '@wordpress/hooks';
import "./style.scss";

// import block.json
import metadata from "./block.json";

// import Pure React Component
import HelloWorld from "./HelloWorld";

// Register React Component in Gutenberg
import { registerFaustBlock } from "@faustwp/block-editor-utils";

registerFaustBlock(HelloWorld, { blockJson: metadata });
```
10. Replace `hello-world-block`'s generated `block.json file with the below code block.

In `app/public/wp-content/plugins/hello-world-block/src/block.json`:
```json
{
	"$schema": "https://schemas.wp.org/trunk/block.json",
	"apiVersion": 2,
	"name": "create-block/hello-world-block",
	"version": "0.1.0",
	"title": "Hello World Block",
	"category": "widgets",
	"icon": "smiley",
	"description": "Example block scaffolded with Create Block tool.",
	"supports": {
		"html": false
	},
	"attributes": {
		"message": {
			"type": "string",
			"default": "Hello World"
		},
		"rangeField": {"type": "number", "default": 0}
	},
	"textdomain": "hello-world-block",
	"editorScript": "file:./index.js",
	"editorStyle": "file:./index.css",
	"style": "file:./style-index.css"
}
```
11. Add a file named `HelloWorld.js` and add the below code block.
```js
// In app/public/wp-content/plugins/hello-world-block/src/HelloWorld.js:

function HelloWorld({ style, className, attributes, children, ...props }) {
	const styles = {
		...style,
		backgroundColor: attributes.bg_color,
		color: attributes.text_color,
	};
	return (
		<div
			{...props}
			style={styles}
			className={className}
			dangerouslySetInnerHTML={{ __html: attributes.message }}
		/>
	);
}

HelloWorld.config = {
	name: "HelloWorld",
        editorFields: {
		message: {
			type: "string",
			default: "",
			label: "My Message",
			location: "editor",
		},
		rangeField: {
			control: "range",
			min: 0,
			max: 50
		}
	},
};
export default HelloWorld;
```
12. Run `npm start` from your plugin's file, `hello-world-block`.

13. After ensuring your block plugin, `Hello World Block` is active.
14. Go into the editor and try to create a new `Hello World Block` block. You should look to the editor and see the rangeField panel visible with whatever value you use. Ensure that upon selecting value that it persists between saves and edits.
15. Ensure that the min, max values are within the right region.

### CODE REVIEW

**Comment by blakewilson** [comment]
Hey @theodesp,

I ran through your testing steps but am not seeing any blocks fields in my Hello World Block:

Screenshot 2023-09-13 at 8 48 52 AM

**Review by blakewilson — APPROVED** [review]
Nice @theodesp! Works great. Was able to confirm locally that the text field and range were editable and were persisted after save:

Screenshot 2023-09-13 at 8 54 58 AM

Screenshot 2023-09-13 at 8 54 53 AM

**Review by blakewilson — CHANGES_REQUESTED** [review]
One thing, let's make sure we get a changset created.

File: packages/block-editor-utils/package.json [hunk 1]
```diff
@@ -7,8 +7,7 @@
   "types": "dist/mjs/index.d.ts",
   "peerDependencies": {
     "react": "^18.0.0",
-    "react-dom": "^18.0.0",
-    "@faustwp/blocks": ">=2.0.0"
```
Comment (blakewilson) [inline packages/block-editor-utils/package.json:11]: Do we no longer need `@faustwp/blocks` as a peer dep? @theodesp

Comment (theodesp) [reply packages/block-editor-utils/package.json:11]: There is no import from this library.

**Review by blakewilson — APPROVED** [review]

**Pull request merged**